core/ir/x86/instr.c

```diff
@@ -1346,21 +1346,18 @@ instr_may_write_zmm_or_opmask_register(instr_t *instr)
 {
     if (instr_get_prefix_flag(instr, PREFIX_EVEX))
         return true;
-    if (instr_raw_bits_valid(instr)) {
+    if (instr_raw_bits_valid(instr) &&
         /* At least 4 for evex bytes + 1 opcode byte. */
-        if (instr_length(NULL, instr) >= 5) {
-            if (instr_get_raw_byte(instr, 0) == EVEX_PREFIX_OPCODE) {
-                if (instr_get_isa_mode(instr) == DR_ISA_AMD64 ||
-                    /* If evex, P[3:2] must be 0 and P[10] must be 1. */
-                    TESTALL(MODRM_BYTE(3, 0, 0), instr_get_raw_byte(instr, 1))) {
-                    if (!TEST(0xC, instr_get_raw_byte(instr, 1)) &&
-                        TEST(0x04, instr_get_raw_byte(instr, 2))) {
-                        return true;
-                    }
-                }
+        instr_length(NULL, instr) >= 5 &&
+        instr_get_raw_byte(instr, 0) == EVEX_PREFIX_OPCODE) {
+        if (instr_get_isa_mode(instr) == DR_ISA_AMD64 ||
```

Comment:
What is this change?  Just collapsing the if's into &&?  But why does it stop here?

Reply:
Seemed slightly more readable to my eyes as the last one was an `||`, but sure I'll collapse this too.

Reply:
Done. Collapsing partially was quite arbitrary on my part. Maybe I didn't think this part through enough.